utopia-api/src/property-controls/property-controls.ts

```diff
@@ -21,71 +21,84 @@ export type BaseControlType =
   | 'vector3'
   | 'vector4'
 
-interface AbstractControlDescription<T extends ControlType> {
+export type CheckboxControlDescription = {
+  control: 'checkbox'
   label?: string
-  control: T
-  defaultValue?: unknown
-  visibleByDefault?: boolean
-}
-
-interface AbstractBaseControlDescription<T extends BaseControlType>
-  extends AbstractControlDescription<T> {}
-
-export interface CheckboxControlDescription extends AbstractBaseControlDescription<'checkbox'> {
   defaultValue?: boolean
+  visibleByDefault?: boolean
   disabledTitle?: string
   enabledTitle?: string
 }
 
-export interface ColorControlDescription extends AbstractBaseControlDescription<'color'> {
+export type ColorControlDescription = {
+  control: 'color'
+  label?: string
+  visibleByDefault?: boolean
   defaultValue?: string
 }
 
 export type AllowedEnumType = string | boolean | number | undefined | null
-export interface BasicControlOption<T> {
+export type BasicControlOption<T> = {
```

Comment:
Why are all these changing from `interface` to `type`?

Reply:
Yeah, please can we stick with `interface`

Reply:
sure thing